aptos-move/block-executor/src/executor.rs

```diff
@@ -1221,15 +1222,42 @@ where
         // This call finalizes the output and may not be concurrent with any other
         // accesses to the output (e.g. querying the write-set, events, etc), as
         // these read accesses are not synchronized and assumed to have terminated.
-        last_input_output.record_materialized_txn_output(
+        let trace = last_input_output.record_materialized_txn_output(
             txn_idx,
             aggregator_v1_delta_writes,
             materialized_resource_write_set
                 .into_iter()
                 .chain(serialized_groups)
                 .collect(),
             materialized_events,
-        )
+        )?;
+
+        if environment.async_runtime_checks_enabled() && !trace.is_empty() {
+            // Note that the trace may be empty (if block was small and Block-STM decides not to
```

Comment:
it's either in block executor (wouldn't even call this block STM), the heuristics below, or it seems also the aptos vm may decide to just run it in place.

Reply:
adjusted the comment